### PR TITLE
Fix bed deletion

### PR DESCRIPTION
**Pull request by idoash4**

### CODE REVIEW

**Comment by ProgrammerDan** [comment]
Looks good to me, @rourke750  or @Maxopoly any issues?

**Comment by rourke750** [comment]
looks good.

File: src/vg/civcraft/mc/bettershards/misc/TeleportInfo.java [hunk 1]
```diff
@@ -42,6 +44,32 @@ public String toString() {
...
+
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+
+		final TeleportInfo other = (TeleportInfo) obj;
+		
+		try {
+			if (!(world.equalsIgnoreCase(other.getWorld())
+					|| (Bukkit.getWorld(UUID.fromString(world)).getName().equalsIgnoreCase(other.getWorld())
```
Comment (Maxopoly) [inline src/vg/civcraft/mc/bettershards/misc/TeleportInfo.java]: This seems like a big NPE source, especially considering that often TeleportInfos are handled, which are not on this server.

File: src/vg/civcraft/mc/bettershards/misc/TeleportInfo.java [hunk 2]
```diff
@@ -42,6 +44,32 @@ public String toString() {
...
+							|| Bukkit.getWorld(UUID.fromString(other.getWorld())).getName().equalsIgnoreCase(world)))) {
+				return false;
+			}
+		} catch (Exception e) {
+			return false;
```
Comment (Maxopoly) [inline src/vg/civcraft/mc/bettershards/misc/TeleportInfo.java]: Just do null checks instead of this

**Comment by Maxopoly** [comment]
@BlackXnt

**Comment by idoash4** [comment]
@Maxopoly @rourke750

**Comment by ProgrammerDan** [comment]
Any chance to rebase?

**Comment by idoash4** [comment]
done

**Pull request merged**